tests/pipeline/test_events.py

```diff
@@ -0,0 +1,191 @@
+            ANNOUNCEMENT_FIELD_NAME,
+        )
+
+    @lazyval
+    def next_announcement_loader(self):
+        return self._previous_event_date_loader(
+            self.dataset.previous_announcement,
+            ANNOUNCEMENT_FIELD_NAME,
+        )
+
+
+class EventDataSetLoaderNoExpectedCols(EventsLoader):
+
+    def __init__(self,
+                 all_dates,
+                 events_by_sid,
+                 infer_timestamps=False,
+                 dataset=EventDataSet):
+        super(EventDataSetLoaderNoExpectedCols, self).__init__(
+            all_dates,
+            events_by_sid,
+            infer_timestamps=infer_timestamps,
+            dataset=dataset,
+        )
+
+
+dtx = pd.date_range('2014-01-01', '2014-01-10')
+
+
+def assert_loader_error(events_by_sid, error, msg, infer_timestamps=True):
+    with assertRaises(error) as context:
+        EventDataSetLoader(
+            dtx, events_by_sid, infer_timestamps=infer_timestamps,
+        )
+        assert msg in context.exception
+
+
+class EventLoaderTestCase(TestCase):
+
+    def test_no_expected_cols_defined(self):
+        events_by_sid = {0: pd.DataFrame({ANNOUNCEMENT_FIELD_NAME: dtx})}
+        assert_loader_error(events_by_sid, TypeError, ABSTRACT_METHODS_ERROR)
+
+    def test_wrong_cols(self):
+        wrong_col_name = 'some_other_col'
+        # Test wrong cols (cols != expected)
+        events_by_sid = {0: pd.DataFrame({wrong_col_name: dtx})}
+        assert_loader_error(
+            events_by_sid, ValueError, WRONG_COLS_ERROR % (
+                EventDataSetLoader.expected_cols, 0, wrong_col_name
+            )
+        )
+
+    @parameterized.expand([
+        # DataFrame without timestamp column and infer_timestamps = True
+        [pd.DataFrame({ANNOUNCEMENT_FIELD_NAME: dtx}), True],
+        # DataFrame with timestamp column
+        [pd.DataFrame({ANNOUNCEMENT_FIELD_NAME: dtx,
+                       TS_FIELD_NAME: dtx}), False],
+        # DatetimeIndex with infer_timestamps = True
+        [pd.DatetimeIndex(dtx, name=ANNOUNCEMENT_FIELD_NAME), True],
+        # Series with DatetimeIndex as index and infer_timestamps = False
+        [pd.Series(dtx, index=dtx, name=ANNOUNCEMENT_FIELD_NAME), False]
+    ])
+    def test_conversion_to_df(self, df, infer_timestamps):
+
+        events_by_sid = {0: df}
+        loader = EventDataSetLoader(
+            dtx,
+            events_by_sid,
+            infer_timestamps=infer_timestamps,
+        )
+        self.assertEqual(
+            loader.events_by_sid.keys(),
+            events_by_sid.keys(),
+        )
+
+        if infer_timestamps:
+            expected = pd.Series(index=[dtx[0]] * 10, data=dtx, )
+        else:
+            expected = pd.Series(index=dtx, data=dtx,)
+        # Check that index by first given date has been added
+        assert_series_equal(
+            loader.events_by_sid[0][ANNOUNCEMENT_FIELD_NAME],
+            expected,
+            check_names=False
```

Comment:
I think we need to check the names of the columns because we use that to do slicing later.